### DISCUSSION TITLE
Turn the deployment script into a Rust-based CLI

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#794] [Ideas]
Just had this idea while reviewing @smol-ninja's PR here:

https://github.com/sablier-labs/v2-core/pull/752

What would you (Shub and @andreivladbrg) say about turning the deployment Bash script into a Rust-based CLI meant for internal use at Sablier?

**Comment by andreivladbrg** [comment]
Actually, my initial version of the script was in Rust, but I gave up on the idea because:
- Initially, the script was not as complex as it is now.
- I couldn't find a suitable place in the project to include it.

So I am pro in making this change

**Reply by PaulRBerg** [reply]
Great, thanks for confirming.

**Comment by smol-ninja** [comment]
I vote for this too.

It would become very organised and safe if we move to Rust. Debugging Bash scripts is a migraine (ex. that string parsing problem was a pain) and it can get really messy for large projects especially if we decide to turn into monorepo.

**Reply by PaulRBerg** [reply]
Awesome. We have a plan. Created a [task on Notion](https://www.notion.so/sablier/16ad272c72f34161b6ab1236e5cf35de?v=94b8c768d3854273a8759b5ba47bd0f7&p=cbb4f741b65f4dd68bc607e67dcb0ade&pm=s).
